lib/matplotlib/axis.py

```diff
@@ -1838,6 +1853,34 @@ def set_ticks(self, ticks, labels=None, *, minor=False, **kwargs):
         choice to prevent the surprise of a non-visible tick. If you need
         other limits, you should set the limits explicitly after setting the
         ticks.
+
+        This method implicitly calls :meth:`.set_major_locator` (or
+        :meth:`.set_minor_locator` if *minor=True*) with a newly-constructed
+        :class:`~matplotlib.ticker.FixedLocator` initialized to the provided
+        tick locations.
+
+        Do not add ticks manually to an automatically ticked axis
+        by e.g. doing (for the *y* axis, in this example)::
+
+            special_ticks = [1.1, 2.5, 3.7] # arbitrary tick locations
+            ticks = axes.get_yticks()
+            ticks_ext = np.append(ticklist, special_ticks)
+            axes.set_yticks(ticks_ext)
+
```

Comment:
Ok, I concede that I put in a lot here. If this is too much, maybe put it in an example titled "Adding custom ticks to an axis", or somewhere else where it would be appropriate, as official guidance for this kind of problem?

In honesty, I got to the doc for `Axes.set_yticks` from some accepted StackOverflow answer that proposes a flawed solution, e.g. [this one in the second part](https://stackoverflow.com/questions/43083464/how-to-append-a-single-labeled-tick-to-x-axis-using-matplotlib) when looking how to approach the task I had in mind.